### PR TITLE
DOC: Remove auto_close option form read_hdf

### PR DESCRIPTION
**Pull request by bashtage**
Remove docstring indicating auto_close can be used in read_hdf.
This value is always ignored.
Also removes unreachable code.

xref #9327

### CODE REVIEW

**Comment by jreback** [comment]
hmm, this should be hit in the iterator, or if you pass an open store to ``read_hdf``. Maybe not tested.

**Comment by bashtage** [comment]
It doesn't have a `return` so I don't think it can't do anything.  Maybe I'm missing the point.  Tests pass with it removed.  

There is also another problem if a buffer is passed - there is no code path for a buffer since the `if` only checks for `basestring` and there is no `else`

**Comment by bashtage** [comment]
I could probably improve it to handle an open store (although why one would want to use `read_hdf` if they have a store open is not obvious).  The path to handle a buffer should be fairly straightforward (for example, if reading a HDF file from the internet).  Presumably no one has ever tried to do this using `read_hdf`

**Comment by jreback** [comment]
I *think* it should handle an open store (and not close it), that is the key. But of its not tested. So for now, let's put a test that hits that path (and you can raise ``NotImplemented`` or make it work), though the object has to be an actual ``HDFStore`` (or then its a ``ValueError``).

Since this hasn't been hit in real code, either it works (maybe) or raises some weird error. Either way its not tested.

**Comment by bashtage** [comment]
It now checks and also raises if a generic buffer type is passed.  

Added test for each.

File: pandas/io/tests/test_pytables.py [hunk 1]
```diff
@@ -4669,6 +4669,49 @@ def test_to_hdf_with_object_column_names(self):
                 result = pd.read_hdf(path, 'df', where="index = [{0}]".format(df.index[0]))
                 assert(len(result))
 
+    def test_read_hdf_open_store(self):
+        df = DataFrame(np.random.rand(4, 5),
```
Comment (jreback) [inline pandas/io/tests/test_pytables.py]: can you put the issues in a comment here

**Comment by jreback** [comment]
lgtm. minor comment ping when green.

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/51047d4992b650bd60c47119ca0f373c61e82fce

**Pull request closed without merging**